cosmwasm/contracts/crosschain-registry/src/execute.rs

```diff
@@ -40,6 +41,102 @@ pub struct ContractAliasInput {
     pub new_alias: Option<String>,
 }
 
+pub fn propose_pfm(
+    ctx: (DepsMut, Env, MessageInfo),
+    chain: String,
+) -> Result<Response, ContractError> {
+    let (deps, env, info) = ctx;
+
+    // enforce lowercase
+    let chain = chain.to_lowercase();
+
+    // validation
+    let registry = Registry::default(deps.as_ref());
+    let coin = cw_utils::one_coin(&info)?;
+    let native_chain = registry.get_native_chain(&coin.denom)?;
+
+    if native_chain.as_ref() != chain {
+        return Err(ContractError::CoinFromInvalidChain {
+            supplied_chain: native_chain.as_ref().to_string(),
+            expected_chain: chain,
+        });
+    }
+
+    // check if the chain is already registered or is in progress
+    if let Some(chain_pfm) = CHAIN_PFM_MAP.may_load(deps.storage, &chain)? {
```

Comment:
Note to self: Since validation is not properly checking the sender anymore, we should restrict who can propose a chain for now